### DISCUSSION TITLE
Multiple instances of Jamulus client existing at the same time on the same machine

### DISCUSSION THREAD

**Discussion by FredNor78** [jamulussoftware/jamulus#1736] [UI Related]
If the Jamulus 3.7.0 client application is already running on my Windows 10 20H2 machine and connected to a server, when I start again Jamulus from the application shortcut on my desktop, it launches an additionnal instance of the client instead of switching to the existing window of the application. As a result, I can see clones of my session as separate users on the same server in the main window. 

Do you think this is appropriate? It looks like nonsense because audio input and output cannot be shared anyway

Could we consider to implement modification in order than instead of launching a separate instance, Jamulus would switch the focus to the existing current instance?

Regards

Fred

**Comment by chrisrimple** [comment]
What is the exact command line for your shortcut?  Did you modify it after installation?  I ask because having multiple shortcuts to the same Jamulus application, but with different INI files parameters specified in the command line, is the intended method for running multiple Jamulus instances on the same Windows computer.  If you've modified the command line at all, that may be why running the shortcut again doesn't just activate the already-running instance.

**Reply by FredNor78** [reply]
Hi Chris!
I use the Jamulus standard entry appearing in the drop list when I strike the WinKey then Jamulus in the search field. And the result is the same when I create on the Desktop a shortcut with the command line ""C:\Program Files\Jamulus\Jamulus.exe" and no additionnal qualifier or parameter

**Comment by dcorson-ticino-com** [comment]
This is desired behavior.
Running multiple instances allows using multiple sound sources/sinks, be they interfaces or DAWs, for example.
Normally one would use the -i command line option to give each instance its own initialization file.
To reopen a minimized instance click on the icon in the task bar, not the application shortcut.

**Reply by FredNor78** [reply]
Understood.
Thanks for the explaination
:)
